UnityProject/Assets/Scripts/Core/Chat/Chat.cs

```diff
@@ -233,16 +233,19 @@ public static void AddChatMsgToChat(ConnectedPlayer sentByPlayer, string message
 			}
 		}
 
-		//Check if there's any items on the player that affects their chat (e.g : headphones, muzzles, etc)
-		foreach (var slots in player.DynamicItemStorage.ServerContents.Values)
+		if (player.IsDeadOrGhost == false)
 		{
-			foreach (var slot in slots)
+			//Check if there's any items on the player that affects their chat (e.g : headphones, muzzles, etc)
+			foreach (var slots in player.DynamicItemStorage.ServerContents.Values)
 			{
-				if (slot.IsEmpty) continue;
-				if (slot.Item.TryGetComponent<IChatInfluncer>(out var listener)
-				    && listener.RunChecks() == true)
+				foreach (var slot in slots)
 				{
-					chatEvent = listener.InfluenceChat(chatEvent);
+					if (slot.IsEmpty) continue;
+					if (slot.Item.TryGetComponent<IChatInfluncer>(out var listener)
```

Comment:
What is a "chat influncer"?

Reply:
A thing that influences the chat going through

So for megaphone's example, when the `InfluenceChat()` function gets called it influences something in the ChatEvent that is being sent. In this case we are influencing the `voiceLevel` of the chat event.

Reply:
It's also an interface not an abstract class

Reply:
So a chat influ**e**ncer then, no?

Reply:
Ah, fixed the type. thanks for pointing that out.